stdlib/public/core/CompilerProtocols.swift

```diff
@@ -338,7 +338,7 @@ public protocol ExpressibleByUnicodeScalarLiteral {
   /// A type that represents a Unicode scalar literal.
   ///
   /// Valid types for `UnicodeScalarLiteralType` are `UnicodeScalar`,
-  /// `String`, and `StaticString`.
+  /// `Character`, `String`, and `StaticString`.
```

Comment:
It's a start, but it doesn't check semantics.  Please write a test using StdlibUnittest that checks the results at runtime, thanks!